cmd/traffic/cmd/manager/internal/state/state.go

```diff
@@ -555,16 +556,28 @@ func (s *State) RemoveIntercept(interceptID string) bool {
 	return didDelete
 }
 
-func (s *State) GetIntercept(interceptID string) *rpc.InterceptInfo {
-	intercept, _ := s.intercepts.Load(interceptID)
-	return intercept
+// GetInterceptAPIKeys returns the s.interceptAPIKeys map.
```

Comment:
This feels dangerous. The map needs to be protected by a mutex but the comment doesn't mention that. Perhaps create a slice that just contains the values within mutex lock and return that instead?